### ISSUE TITLE
[accordion] `Accordion.Trigger` custom `render` + `nativeButton={false}` breaks semantics and keyboard nav

### ISSUE THREAD

**Issue by ZeeshanTamboli** [mui/base-ui#3632]
# Bug report

## Current behavior

When using `Accordion.Trigger` with the `render` prop and `nativeButton={false}`, the documentation suggests rendering a non-button element (e.g. `<div>`):

> Set to false if the rendered element is not a button (e.g. div).

https://base-ui.com/react/components/accordion#trigger

However, `Accordion.Trigger` is expected to be used inside `Accordion.Header`, which renders an `<h3>` element.

According to HTML specifications, any heading element may only contain phrasing content and cannot contain a `<div>`. This leads to invalid HTML when following the documented guidance.

Additionally, when rendering a non-button valid element (e.g. `<span>`) with `nativeButton={false}`, keyboard navigation (Arrow up down keys, Home, End) does not work out of the box due to querySelector only selecting active triggers with `type="button"`. It should also select with `role="button"`.

## Expected behavior

- The docs should not mention `div` inside `h3` when using `render` + `nativeButton={false}`. 
- Keyboard navigation should work if a custom trigger element is used for e.g. `span`.

## Reproducible example

Link to docs: https://base-ui.com/react/components/accordion#trigger

Link to StackBlitz: https://stackblitz.com/edit/t6iymq5o?file=src%2FApp.tsx

Screen recording for navigation:

https://github.com/user-attachments/assets/cd36857a-5934-4f43-841a-73b469622943

## Base UI version

`latest`

## Which browser are you using?

Chrome

## Which OS are you using?

Windows

**Comment by ZeeshanTamboli** [comment]
I can create a PR to update the docs and do a fix for navigation once confirmed as a bug.

**Comment by atomiks** [comment]
It says `<div>` since `NativeButtonProps` has the same JSDoc description globally. It would need to be edited in cases where invalid HTML gets created (which is fine)

> does not work out of the box due to querySelector only selecting active triggers with type="button". It should also select with role="button".

> I can create a PR to update the docs and do a fix for navigation once confirmed as a bug.


Yes, seems like it needs a fix
